src/rdkafka_cgrp.c

```diff
@@ -3865,99 +4239,164 @@ rd_kafka_error_t *rd_kafka_consumer_group_metadata_write (
-                return rd_kafka_error_new(
-                        RD_KAFKA_RESP_ERR__BAD_MSG,
-                        "Input buffer is not a serialized "
-                        "consumer group metadata object");
-
+                return rd_kafka_error_new(RD_KAFKA_RESP_ERR__BAD_MSG,
+                                          "Input buffer is not a serialized "
+                                          "consumer group metadata object");
         memcpy(&generation_id, buf+magic_len, generationid_len);
 
         group_id = buf + magic_len + generationid_len;
+        next = str_is_printable(group_id, end);
+        if (!next)
+                return rd_kafka_error_new(RD_KAFKA_RESP_ERR__BAD_MSG,
+                                          "Input buffer group id is not safe");
 
-        /* Check that group_id is safe */
-        for (s = group_id ; s < end - 1 ; s++) {
-                if (!isprint((int)*s))
-                        return rd_kafka_error_new(
-                                RD_KAFKA_RESP_ERR__BAD_MSG,
-                                "Input buffer group id is not safe");
+        member_id = next;
+        next = str_is_printable(member_id, end);
+        if (!next)
+                return rd_kafka_error_new(RD_KAFKA_RESP_ERR__BAD_MSG,
+                                          "Input buffer member id is not "
+                                          "safe");
+
+        group_instance_id_is_null = (int8_t)*next++;
+        if (!group_instance_id_is_null) {
+                group_instance_id = next;
+                next = str_is_printable(group_instance_id, end);
+                if (!next)
+                        return rd_kafka_error_new(RD_KAFKA_RESP_ERR__BAD_MSG,
+                                                  "Input buffer group "
+                                                  "instance id is not safe");
         }
 
-        if (*s != '\0')
-                return rd_kafka_error_new(
-                        RD_KAFKA_RESP_ERR__BAD_MSG,
-                        "Input buffer has invalid stop byte");
+        if (next != end)
+                return rd_kafka_error_new(RD_KAFKA_RESP_ERR__BAD_MSG,
+                                          "Input buffer bad length");
 
-        /* We now know that group_id is printable-safe and is nul-terminated */
-        *cgmdp = rd_kafka_consumer_group_metadata_new_with_genid(group_id,
-                                                                 generation_id);
+        *cgmdp = rd_kafka_consumer_group_metadata_new_with_genid(
+                                                        group_id,
+                                                        generation_id,
+                                                        member_id,
+                                                        group_instance_id);
 
         return NULL;
 }
 
 
-static int unittest_consumer_group_metadata (void) {
+static int unittest_iteration(const char *group_id,
```

Comment:
unittest_consumer_group_metadata_iteration() or at least something that gives this a contextual name.
And ut_ is fine for static unittest funciton prefixes.

Reply:
apologies then for > 80 col